src/main/java/ee/tuleva/onboarding/auth/ocsp/OCSPRequest.java

```diff
@@ -0,0 +1,155 @@
+package ee.tuleva.onboarding.auth.ocsp;
+
+import com.codeborne.security.AuthenticationException;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateHolder;
+import org.bouncycastle.cert.ocsp.*;
+import org.bouncycastle.operator.DigestCalculator;
+import org.bouncycastle.operator.DigestCalculatorProvider;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaDigestCalculatorProviderBuilder;
+
+import java.io.*;
+import java.math.BigInteger;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+@Slf4j
+public class OCSPRequest {
+    private final String ocspServer;
+    private final String ca;
+    private final String method;
+
+    public OCSPRequest(String caToCheck, String ocspServer) {
+        this.ca = caToCheck;
+        this.ocspServer = ocspServer;
+        this.method = "POST";
+    }
+
+    public String checkCertificate(X509Certificate certificate){
+        if(hasCertificateExpired(certificate)){
+            return "CERTIFICATE_EXPIRED";
+        }
+        return checkSerialNumber(certificate.getSerialNumber());
+    }
+
+    private boolean hasCertificateExpired(X509Certificate certificate) {
+        return System.currentTimeMillis() > certificate.getNotAfter().getTime();
+    }
+
+    public String checkSerialNumber(BigInteger serialNumber) {
+
+        String checkedResponse = null;
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+
+        try {
+            log.info("Generating and sending OCSPRequest");
+            OCSPResp response = getOCSPResponse(serialNumber);
+
+            log.info("Validating OCSPResponse");
+            checkedResponse = validateOCSPResponse(response);
+
+            log.info("OCSPResponse validated");
+        } catch (OCSPException | IOException e) {
+            log.warn("Couldn't validate serial number");
+            throw new AuthenticationException(AuthenticationException.Code.UNABLE_TO_TEST_USER_CERTIFICATE, "Couldn't validate serial number", e);
+        }
+
+        return checkedResponse;
+    }
+
+
+    private OCSPResp getOCSPResponse (BigInteger serialNumber) throws IOException {
+        OCSPReq request = generateOCSPRequest(serialNumber);
+        OCSPResp response = sendPost(request.getEncoded());
+        return response;
+    }
+
+
+    private String validateOCSPResponse(OCSPResp response) throws OCSPException {
+
+        String status = "UNABLE_TO_TEST_USER_CERTIFICATE";
+        switch (response.getStatus()) {
+            case 0:
```

Comment:
Magic number. Can we extract well-named constants for the `case 0` and `case 1 ` to make it easier to understand? I.e. what does response status 0 or 1 mean.